src/head_align_tail_dist.py

```diff
@@ -181,7 +190,11 @@ def head_align_tail(prefix, alnm_ext, mode):
                 aligned_ref = alnm.reference_length
                 read_len_total = alnm.infer_read_length()
                 middle = alnm.query_alignment_length
-                head, tail = get_head_tail(alnm.cigartuples)
+                if mode == "transcriptome" and (read in dict_genome_alnm_info):
+                    genome_alnm = dict_genome_alnm_info[read]
+                    head, tail = get_head_tail(genome_alnm.cigartuples)
```

Comment:
The head and tail should be the minimum observed in the sequence's transcriptome and genome alignment:
```
head_g, tail_g = get_head_tail(genome_alnm.cigartuples)
head_t, tail_t = get_head_tail(alnm.cigartuples)
head = min(head_g, head_t)
tail = min(tail_g, tail_t)
```